plugins/utils/yanky.nix

```
@@ -68,70 +174,83 @@ in
             cursor or content changed.
           '';
 
-      ignoreRegisters = helpers.defaultNullOpts.mkListOf types.str [ "_" ] ''
+      ignore_registers = helpers.defaultNullOpts.mkListOf types.str [ "_" ] ''
         Define registers to be ignored.
         By default the black hole register is ignored.
       '';
 
-      updateRegisterOnCycle = helpers.defaultNullOpts.mkBool false ''
+      update_register_on_cycle = helpers.defaultNullOpts.mkBool false ''
         If true, when you cycle through the ring, the contents of the register used to update will
         be updated with the last content cycled.
       '';
     };
 
     picker = {
       select = {
-        action = helpers.mkNullOrOption (with types; either helpers.nixvimTypes.rawLua str) ''
-          This define the action that should be done when selecting an item in the
-          `vim.ui.select` prompt.
-          If you let this option to `null`, this will use the default action: put selected item
-          after cursor.
-
-          This is either:
-          - a `rawLua` value (`action.__raw = "function() foo end";`).
-          - a string. In this case, Nixvim will automatically interpret it as a builtin yanky
-          action.
-            Example: `action = "put('p')";` will translate to
-              `action = require('yanky.picker').actions.put('p')` in lua.
-        '';
+        action = helpers.defaultNullOpts.mkLuaFn' {
+          pluginDefault = null;
+          description = ''
+            This define the action that should be done when selecting an item in the
+            `vim.ui.select` prompt.
+            If you let this option to `null`, this will use the default action: put selected item
+            after cursor.
+          '';
+          example = "require('yanky.picker').actions.put('gp')";
+        };
       };
 
       telescope = {
-        enable = mkEnableOption "the `yank_history` telescope picker.";
+        use_default_mappings = helpers.defaultNullOpts.mkBool true ''
+          This define or overrides the mappings available in Telescope.
 
-        useDefaultMappings = helpers.defaultNullOpts.mkBool true ''
-          This define if default Telescope mappings should be used.
+          If you set this option to `true`, mappings will be merged with default mappings.
         '';
 
-        mappings = helpers.mkNullOrOption (with types; attrsOf (attrsOf str)) ''
-          This define or overrides the mappings available in Telescope.
+        mappings = helpers.defaultNullOpts.mkAttrsOf' {
+          type = with helpers.nixvimTypes; either strLuaFn (attrsOf strLuaFn);
+          apply =
+            mappings:
+            helpers.ifNonNull' mappings (
+              mapAttrs (
+                _: v:
+                if
+                  isString v
+                # `mappings.default` is a lua function
+                then
```

I think the comment's position is messing with nixfmt. Moving the comment before or after should work:

```suggestion
                if isString v then
                  # `mappings.default` is a lua function
```